src/optimize/dynamic_dll_plugin/dll_compiler.js

```diff
@@ -197,18 +217,54 @@ export class DllCompiler {
         }));
 
         if (webpackErrors) {
-          this.logWithMetadata(
-            ['fatal', 'optimize:dynamic_dll_plugin'],
-            `Client vendors dll compilation failed`
-          );
+          // Reject with webpack fatal errors
           return reject(webpackErrors);
         }
 
-        // Otherwise let it proceed
-        this.logWithMetadata(
-          ['info', 'optimize:dynamic_dll_plugin'],
-          `Client vendors dll compilation finished with success`
-        );
+        // Identify if we have not allowed modules
+        // bundled inside the dll bundle
+        const notAllowedModules = [];
+
+        stats.compilation.modules.forEach((module) => {
+          // ignore if no module or userRequest are defined
+          if (!module || !module.resource) {
+            return;
+          }
+
+          // ignore if this module represents the
+          // dll entry file
+          if (module.resource === this.getEntryPath()) {
+            return;
+          }
+
+          // A module is not allowed if it's not a node_module, a webpackShim
+          // or the reasons from being bundled into the dll are not node_modules
+          if(notInNodeModulesOrWebpackShims(module.resource)) {
+            const reasons = module.reasons || [];
+
+            reasons.forEach((reason) => {
+              // Skip if we can't read the reason info
+              if (!reason || !reason.module || !reason.module.resource) {
+                return;
+              }
+
+              // Is the reason for this module being bundle a
+              // node_module or no?
+              if (notInNodeModules(reason.module.resource)) {
+                notAllowedModules.push(module.resource);
+              }
+            });
+          }
+        });
+
+        if (notAllowedModules.length) {
+          // Delete the built dll, as it contains invalid modules, and reject listing
+          // all the not allowed modules
+          await rimrafAsync(this.rawDllConfig.outputPath);
```

Comment:
I'm worried about turning this callback into an async function since nothing is catching the rejected promises. Maybe it's best to use `rimraf.sync()` instead

Reply:
Good find @spalger 👍 
Maybe we can just do the following to keep the current error handling behaviour:

```javascript
if (notAllowedModules.length) {
  // Delete the built dll, as it contains invalid modules, and reject listing
  // all the not allowed modules
  try {
    await rimrafAsync(this.rawDllConfig.outputPath);
  } catch (e) {
    return reject(e);
  }

  return reject(`The following modules are not allowed to be bundled into the dll: \n${notAllowedModules.join('\n')}`);
}
```

What do u think?